src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1316,11 +1498,24 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::GetWithParam(spinel_prop
 
     assert(mWaitingTid == 0);
 
-    mPropertyFormat = aFormat;
     va_start(mPropertyArgs, aFormat);
-    error = Request(true, SPINEL_CMD_PROP_VALUE_GET, aKey, SPINEL_DATATYPE_DATA_S, aParam, aParamSize);
+    error =
+        RequestWithPropertyFormat(aFormat, SPINEL_CMD_PROP_VALUE_GET, aKey, SPINEL_DATATYPE_DATA_S, aParam, aParamSize);
     va_end(mPropertyArgs);
-    mPropertyFormat = nullptr;
+
+#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
+    if (mRcpFailed)
+    {
+        RecoverFromRcpFailure();
+
+        va_start(mPropertyArgs, aFormat);
+        error = RequestWithPropertyFormat(aFormat, SPINEL_CMD_PROP_VALUE_GET, aKey, SPINEL_DATATYPE_DATA_S, aParam,
+                                          aParamSize);
+        va_end(mPropertyArgs);
+
+        VerifyOrDie(!mRcpFailed, OT_EXIT_FAILURE);
+    }
+#endif
```

Comment:
Can we use code like below?

```c++

#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
    for (uint8_t i = 0; i <= OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT; i++)
    {
#endif

        va_start(mPropertyArgs, aFormat);
        error = RequestWithPropertyFormat(aFormat, SPINEL_CMD_PROP_VALUE_GET, aKey, SPINEL_DATATYPE_DATA_S, aParam,
                                          aParamSize);
        va_end(mPropertyArgs);

#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
        if (mRcpFailed)
        {
            RecoverFromRcpFailure();
        } else {
            break;
        }
    }

    VerifyOrDie(!mRcpFailed, OT_EXIT_FAILURE);
#endif
```
The code above is just for demonstration. But it shows that it's possible to somehow reuse the code between `va_start` and `va_end`. 

furthermore, I think it's incorrect to just `VerifyOrDie(!mRcpFailed, OT_EXIT_FAILURE);` after second `Get` or `Set`.

Reply:
Agreed. But for `OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT=n` it means we allow at most `n` consecutive failures in a single call to `RecoverFromRcpFailure()`, if we allow `n` calls to `RecoverFromRcpFailure()` then it's at most `n^2` failures in a set/get operation.

I think it's better to change the loop to a `while` loop so that it:

Keeps retrying get/set until one of the following is satisfied:
1. Set/get succeeds without failure.
1. More than `n` failures happen in a single recovery process.

Thoughts?

Reply:
Did the change in <https://github.com/openthread/openthread/pull/5694/commits/3b8c25b0a42d6720a91be1e8267ff5019e0327f9>. You can see if it's OK.

Reply:
I am a little bit worried since the new code seems to introduce a potential dead loop.
Nevertheless, I don't see a obvious way to handle this.

Reply:
If wrong logic causes a dead loop, I think it will be very obvious, and caught by CI tests.

Reply:
SGTM